packages/dev/core/src/Physics/v2/ragdoll.ts

```diff
@@ -0,0 +1,331 @@
+
+                // First define the box dimensions, so we can then use them when calling CreateBox().
+                const currentRagdollBoneProperties: RagdollBoneProperties = {
+                    width: this._config[i].width,
+                    depth: this._config[i].depth,
+                    height: this._config[i].height,
+                    size: this._config[i].size,
+                };
+
+                const box = MeshBuilder.CreateBox(boneNames[ii] + "_box", currentRagdollBoneProperties, this._scene);
+
+                box.visibility = 0;
+
+                // Define the rest of the box properties.
+                currentRagdollBoneProperties.joint = config[i].joint !== undefined ? config[i].joint : this._defaultJoint;
+                currentRagdollBoneProperties.rotationAxis = config[i].rotationAxis !== undefined ? config[i].rotationAxis : Axis.X;
+                currentRagdollBoneProperties.min = config[i].min !== undefined ? config[i].min : this._defaultJointMin;
+                currentRagdollBoneProperties.max = config[i].max !== undefined ? config[i].max : this._defaultJointMax;
+
+                // Offset value.
+                let boxOffset = 0;
+                if ((config[i].putBoxInBoneCenter !== undefined && config[i].putBoxInBoneCenter) || this._putBoxesInBoneCenter) {
+                    if (currentBone.length === undefined) {
+                        console.log("The length property is not defined for bone", currentBone.name);
+                    }
+                    boxOffset = currentBone.length / 2;
+                } else if (config[i].boxOffset !== undefined) {
+                    boxOffset = config[i].boxOffset;
+                }
+                currentRagdollBoneProperties.boxOffset = boxOffset;
+
+                // Offset axis.
+                const boneOffsetAxis = config[i].boneOffsetAxis !== undefined ? config[i].boneOffsetAxis : this._boneOffsetAxis;
+                const boneDir = currentBone.getDirection(boneOffsetAxis, this._mesh);
+                currentRagdollBoneProperties.boneOffsetAxis = boneOffsetAxis;
+
+                box.position = currentBone.getAbsolutePosition(this._mesh).add(boneDir.scale(boxOffset));
+
+                const mass = config[i].mass !== undefined ? config[i].mass : this._mass;
+                const restitution = config[i].restitution !== undefined ? config[i].restitution : this._restitution;
+                const aggregate = new PhysicsAggregate(
+                    box,
+                    PhysicsShapeType.BOX,
+                    {
+                        mass: mass,
+                        restitution: restitution,
+                        friction: 0.6,
+                    },
+                    this._scene
+                );
+                aggregate.body.setCollisionCallbackEnabled(true);
+                aggregate.body.disablePreStep = false;
+                (aggregate.body as any).collisionArray = new Array<number>(1);

```

Comment:
why the cast? is it not available on the class? Should it be added?

Reply:
Fixed.